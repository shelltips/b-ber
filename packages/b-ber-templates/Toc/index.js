<<<<<<< HEAD
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _path = require('path');

var _path2 = _interopRequireDefault(_path);

var _vinyl = require('vinyl');

var _vinyl2 = _interopRequireDefault(_vinyl);

var _utils = require('@canopycanopycanopy/b-ber-lib/utils');

var _State = require('@canopycanopycanopy/b-ber-lib/State');

var _State2 = _interopRequireDefault(_State);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Toc = function () {
    function Toc() {
        (0, _classCallCheck3.default)(this, Toc);
    }

    (0, _createClass3.default)(Toc, null, [{
        key: 'document',
        value: function document() {
            return new _vinyl2.default({
                path: 'toc.document.tmpl',
                contents: new Buffer(_State2.default.templates.dynamicPageHead() + '\n                <nav id="toc" epub:type="toc">\n                    <h2>Table of Contents</h2>\n                    {% body %}\n                </nav>\n                ' + _State2.default.templates.dynamicPageTail() + '\n            ')
            });
        }
    }, {
        key: 'item',
        value: function item(data) {
            return '<a href="' + _path2.default.basename(data.relativePath) + '.xhtml">' + (0, _utils.escapeHTML)((0, _utils.getTitleOrName)(data)) + '</a>';
        }
    }, {
        key: 'items',
        value: function items(data) {
            return '\n            <ol>\n                ' + data.map(function (a) {
                if (a.in_toc === false) return '';
                return '\n                        <li>\n                            ' + Toc.item(a) + '\n                            ' + (a.nodes && a.nodes.length ? Toc.items(a.nodes) : '') + '\n                        </li>\n                    ';
            }).join('') + '\n            </ol>\n        ';
        }
    }]);
    return Toc;
}();

exports.default = Toc;
||||||| parent of 832bbc7... Restructuring
=======
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _path = require('path');

var _path2 = _interopRequireDefault(_path);

var _vinyl = require('vinyl');

var _vinyl2 = _interopRequireDefault(_vinyl);

var _utils = require('@canopycanopycanopy/b-ber-lib/utils');

var _State = require('@canopycanopycanopy/b-ber-lib/State');

var _State2 = _interopRequireDefault(_State);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Toc = function () {
    function Toc() {
        (0, _classCallCheck3.default)(this, Toc);
    }

    (0, _createClass3.default)(Toc, null, [{
        key: 'document',
        value: function document() {
            return new _vinyl2.default({
                path: 'toc.document.tmpl',
                contents: new Buffer('\n                ' + _State2.default.templates.dynamicPageHead() + '\n                <nav id="toc" epub:type="toc">\n                    <h2>Table of Contents</h2>\n                    {% body %}\n                </nav>\n                ' + _State2.default.templates.dynamicPageTail() + '\n            ')
            });
        }
    }, {
        key: 'item',
        value: function item(data) {
            return '<a href="' + _path2.default.basename(data.relativePath) + '.xhtml">' + (0, _utils.escapeHTML)((0, _utils.getTitleOrName)(data)) + '</a>';
        }
    }, {
        key: 'items',
        value: function items(data) {
            return '\n            <ol>\n                ' + data.map(function (a) {
                if (a.in_toc === false) return '';
                return '\n                        <li>\n                            ' + Toc.item(a) + '\n                            ' + (a.nodes && a.nodes.length ? Toc.items(a.nodes) : '') + '\n                        </li>\n                    ';
            }).join('') + '\n            </ol>\n        ';
        }
    }]);
    return Toc;
}();

<<<<<<< HEAD
exports.default = Toc;
/*
const tocItem = list => {
    function render(items) {
        return `
            <ol>
                ${items.map(a => // eslint-disable-line no-confusing-arrow
                    (a.in_toc === false)
                    ? ''
                    : `<li>
                        <a href="${path.basename(a.relativePath)}.xhtml">${escapeHTML(getTitleOrName(a))}</a>
                            ${a.nodes && a.nodes.length ? render(a.nodes) : ''}
                        </li>`
                ).join('')}
            </ol>`
    }

    return render(list)
}*/
>>>>>>> 832bbc7... Restructuring
||||||| parent of 3c03362... Refactoring; Builds
exports.default = Toc;
/*
const tocItem = list => {
    function render(items) {
        return `
            <ol>
                ${items.map(a => // eslint-disable-line no-confusing-arrow
                    (a.in_toc === false)
                    ? ''
                    : `<li>
                        <a href="${path.basename(a.relativePath)}.xhtml">${escapeHTML(getTitleOrName(a))}</a>
                            ${a.nodes && a.nodes.length ? render(a.nodes) : ''}
                        </li>`
                ).join('')}
            </ol>`
    }

    return render(list)
}*/
=======
exports.default = Toc;
>>>>>>> 3c03362... Refactoring; Builds
