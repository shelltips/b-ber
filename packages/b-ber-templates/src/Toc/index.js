/* eslint-disable indent */

import path from 'path'
import File from 'vinyl'
import {Html} from '@canopycanopycanopy/b-ber-lib'
import {getTitleOrName} from '@canopycanopycanopy/b-ber-lib/utils'
import state from '@canopycanopycanopy/b-ber-lib/State'


class Toc {
    static document() {
        return new File({
            path: 'toc.document.tmpl',
            contents: new Buffer(`${state.templates.dynamicPageHead()}
                <nav id="toc" epub:type="toc">
                    <h2>Table of Contents</h2>
                    {% body %}
                </nav>
                ${state.templates.dynamicPageTail()}
            `),
        })
    }
    static item(data) {
        return `<a href="${path.basename(data.relativePath)}.xhtml">${Html.escape(getTitleOrName(data))}</a>`
    }

    static items(data) {
        return `
            <ol>
                ${data.map(a => {
                    if (a.in_toc === false) return ''
                    return `
                        <li>
                            ${Toc.item(a)}
                            ${a.nodes && a.nodes.length ? Toc.items(a.nodes) : ''}
                        </li>
                    `
                    }).join('')
                }
            </ol>
        `
    }
}

export default Toc
