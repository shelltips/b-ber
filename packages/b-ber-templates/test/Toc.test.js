
import Toc from '../src/Toc'

jest.mock('@canopycanopycanopy/b-ber-lib/State', () => ({
    templates: {
        dynamicPageHead: () => {},
        dynamicPageTail: () => {},
    },
}))

describe('templates.Toc', () => {

    it('creates an anchor element', () => {
        const node = {in_toc: true, relativePath: 'test', title: 'Test'}
        expect(Toc.item(node)).toMatchSnapshot()
    })

    it('creates an xhtml document', () => {
        expect(Toc.document().contents).toMatchSnapshot()
        expect(Toc.document().history).toMatchSnapshot()
    })

    it('creates a nested list', () => {
        const node = {in_toc: true, relativePath: 'test', title: 'Test'}
        const nodes = [{in_toc: true, relativePath: 'test-child', title: 'Test Child'}]
        const data = [{
            in_toc: false,
        }, {
            ...node,
            nodes,
        }]
        expect(Toc.items(data)).toMatchSnapshot()
    })

})
