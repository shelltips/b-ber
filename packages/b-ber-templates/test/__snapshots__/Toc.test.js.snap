// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates.Toc creates a nested list 1`] = `
"
            <ol>
                
                        <li>
                            <a href=\\"test.xhtml\\">Test</a>
                            
            <ol>
                
                        <li>
                            <a href=\\"test-child.xhtml\\">Test Child</a>
                            
                        </li>
                    
            </ol>
        
                        </li>
                    
            </ol>
        "
`;

exports[`templates.Toc creates an anchor element 1`] = `"<a href=\\"test.xhtml\\">Test</a>"`;

exports[`templates.Toc creates an xhtml document 1`] = `
Object {
  "data": Array [
    117,
    110,
    100,
    101,
    102,
    105,
    110,
    101,
    100,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    110,
    97,
    118,
    32,
    105,
    100,
    61,
    34,
    116,
    111,
    99,
    34,
    32,
    101,
    112,
    117,
    98,
    58,
    116,
    121,
    112,
    101,
    61,
    34,
    116,
    111,
    99,
    34,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    104,
    50,
    62,
    84,
    97,
    98,
    108,
    101,
    32,
    111,
    102,
    32,
    67,
    111,
    110,
    116,
    101,
    110,
    116,
    115,
    60,
    47,
    104,
    50,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    123,
    37,
    32,
    98,
    111,
    100,
    121,
    32,
    37,
    125,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    47,
    110,
    97,
    118,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    117,
    110,
    100,
    101,
    102,
    105,
    110,
    101,
    100,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
  ],
  "type": "Buffer",
}
`;

exports[`templates.Toc creates an xhtml document 2`] = `
Array [
  "toc.document.tmpl",
]
`;
